lib/Doctrine/Common/Cache/ArrayCache.php

```diff
@@ -59,8 +92,8 @@ protected function doContains($id)
      */
     protected function doSave($id, $data, $lifeTime = 0)
     {
-        $this->data[$id] = $data;
-
+        $ttl = $lifeTime > 0 ? time()+$lifeTime : false;
+        $this->data[$id] = array('content'=>$data, 'ttl'=>$ttl);
```

Comment:
this should use numeric indexes instead of `content` and `ttl` to save memory. or use different arrays for data and ttl.